src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5707,9 +5712,67 @@ private static BoundExpression CreatePlaceholderIfNecessary(BoundExpression expr
         public override BoundNode? VisitCall(BoundCall node)
         {
             // Note: we analyze even omitted calls
-            TypeWithState receiverType = VisitCallReceiver(node);
-            ReinferMethodAndVisitArguments(node, receiverType);
+            if (node.ReceiverOpt is BoundCall receiver)
+            {
+                var calls = ArrayBuilder<BoundCall>.GetInstance();
+
+                calls.Push(node);
+                node = receiver;
+
+                bool originalExpressionIsRead = _expressionIsRead;
+                _expressionIsRead = true;
+
+                while (node.ReceiverOpt is BoundCall receiver2)
+                {
+                    TakeIncrementalSnapshot(node); // Visit does this before visiting each node
+                    calls.Push(node);
+                    node = receiver2;
+                }
+
+                TakeIncrementalSnapshot(node); // Visit does this before visiting each node
+                TypeWithState receiverType = visitAndCheckReceiver(node);
+
+                while (true)
+                {
+                    ReinferMethodAndVisitArguments(node, receiverType);
+
+                    receiver = node;
+                    if (!calls.TryPop(out node!))
+                    {
+                        break;
+                    }
+
+                    VisitExpressionWithoutStackGuardEpilogue(receiver); // VisitExpressionWithoutStackGuard does this after visiting each node
+                    VisitRvalueEpilogue(receiver); // VisitRvalue does this after visiting each node
+
+                    receiverType = ResultType;
+                    CheckCallReceiver(receiver, receiverType, node.Method);
+                }
+
+                _expressionIsRead = originalExpressionIsRead;
+
+                calls.Free();
+            }
+            else
+            {
+                TypeWithState receiverType = visitAndCheckReceiver(node);
```

Comment:
Why don't we need to update/track the expressionIsRead value when dealing with a single invocation?

Reply:
> Why don't we need to update/track the expressionIsRead value when dealing with a single invocation?

Because the calling Visit does that. Above we are setting it for nested calls.